sdk/identity/test-resources-post.ps1

```diff
@@ -0,0 +1,171 @@
+mvn clean install -DskipTests "-Drevapi.skip" "-Dcheckstyle.skip" "-Dspotbugs.skip" "-Dmaven.javadoc.skip=true" "-Dspotless.check.skip=true" -f $azIdentityRootPom | Write-Host
+mvn clean install -DskipTests "-Drevapi.skip" "-Dcheckstyle.skip" "-Dspotbugs.skip" "-Dmaven.javadoc.skip=true" "-Dspotless.check.skip=true" -f "$azStorageRootPom/azure-storage-common/pom.xml" | Write-Host
+mvn clean install -DskipTests "-Drevapi.skip" "-Dcheckstyle.skip" "-Dspotbugs.skip" "-Dmaven.javadoc.skip=true" "-Dspotless.check.skip=true" -f "$azStorageRootPom/azure-storage-internal-avro/pom.xml" | Write-Host
+mvn clean install -DskipTests "-Drevapi.skip" "-Dcheckstyle.skip" "-Dspotbugs.skip" "-Dmaven.javadoc.skip=true" "-Dspotless.check.skip=true" -f "$azStorageRootPom/azure-storage-blob/pom.xml" | Write-Host
+
+
+mvn clean install -DskipTests -f $webappRootPom | Write-Host
+az webapp deploy --resource-group $(getVariable('IDENTITY_RESOURCE_GROUP')) --name $(getVariable('IDENTITY_WEBAPP_NAME')) --src-path "$webappRoot/target/identity-test-webapp-0.0.1-SNAPSHOT.jar" --type jar
+
+Write-Host "Building Function App"
+
+# build function app
+mvn clean package "-DfunctionAppName=$(getVariable('IDENTITY_FUNCTION_NAME'))" "-DresourceGroup=$(getVariable('IDENTITY_RESOURCE_GROUP'))" "-DappServicePlanName=$(getVariable('IDENTITY_APPSERVICE_NAME'))" -f $funcAppPom | Write-Host
+compress-archive  "$funcAppRoot\target\azure-functions\$(getVariable('IDENTITY_FUNCTION_NAME'))\*" -DestinationPath "$funcAppRoot/target/funcpackage.zip"
+az functionapp deployment source config-zip -g $(getVariable('IDENTITY_RESOURCE_GROUP')) -n $(getVariable('IDENTITY_FUNCTION_NAME')) --src "$funcAppRoot/target/funcpackage.zip"
+
+Write-Host "Building VM App"
+# build VM app
+mvn clean package -f "$vmRoot/pom.xml" | Write-Host
+
+# Virtual machine setup
+$vmScript = @"
+sudo apt update && sudo apt install openjdk-8-jdk -y --no-install-recommends
+"@
+az vm run-command invoke -n $DeploymentOutputs['IDENTITY_VM_NAME'] -g $DeploymentOutputs['IDENTITY_RESOURCE_GROUP'] --command-id RunShellScript --scripts "$vmScript"
+
+Write-Host "Getting storage account details"
+$key=az storage account keys list --account-name $DeploymentOutputs['IDENTITY_STORAGE_NAME_1'] --resource-group $DeploymentOutputs['IDENTITY_RESOURCE_GROUP'] --query "[0].value" --output tsv
+
+Write-Host "Creating storage container"
+az storage container create --name "vmcontainer" --account-name $DeploymentOutputs['IDENTITY_STORAGE_NAME_1'] --account-key $key --public-access blob
+
+Write-Host "Uploading file to storage"
+az storage blob upload --container-name "vmcontainer" --file "$vmRoot/target/identity-test-vm-1.0-SNAPSHOT-jar-with-dependencies.jar" --name "testfile.jar" --account-name $DeploymentOutputs['IDENTITY_STORAGE_NAME_1'] --account-key $key
+
+
+if ($IsMacOS -eq $true) {
+  # Not supported on MacOS agents
+  az logout
```

Comment:
If logging out is an important step let's put it inside a [`trap`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_trap?view=powershell-7.4) instead?

Reply:
[This](https://github.com/Azure/azure-sdk-for-java/blob/2426f1e06d4ae027a0c2fb1525fd428a39585e0a/eng/common/TestResources/Remove-TestResources.ps1#L79) is a usage of `trap` in our current codebase.

Reply:
added it